deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java

```diff
@@ -217,22 +220,28 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
     return false;
   }
 
-  private static List<String> enumerateCustomDeepLinks(Element element,
-      Map<Element, String[]> prefixesMap) {
+  private List<DeepLinkAnnotatedElement> generateCustomDeepLinkAnnotatedElements(Element element,
+      DeepLinkEntry.Type type, Map<Element, String[]> prefixesMap) {
+
+    List<DeepLinkAnnotatedElement> deepLinkElements = new ArrayList<>();
     Set<? extends AnnotationMirror> annotationMirrors =
-        AnnotationMirrors.getAnnotatedAnnotations(element, DEEP_LINK_SPEC_CLASS);
-    final List<String> deepLinks = new ArrayList<>();
+            AnnotationMirrors.getAnnotatedAnnotations(element, DEEP_LINK_SPEC_CLASS);
     for (AnnotationMirror customAnnotation : annotationMirrors) {
+
       List<? extends AnnotationValue> suffixes =
-          asAnnotationValues(AnnotationMirrors.getAnnotationValue(customAnnotation, "value"));
-      String[] prefixes = prefixesMap.get(customAnnotation.getAnnotationType().asElement());
-      for (String prefix : prefixes) {
+              asAnnotationValues(AnnotationMirrors.getAnnotationValue(customAnnotation, "value"));
+      String[] prefixesArray = prefixesMap.get(customAnnotation.getAnnotationType().asElement());
+      try {
         for (AnnotationValue suffix : suffixes) {
-          deepLinks.add(prefix + suffix.getValue());
+          deepLinkElements.add(new DeepLinkAnnotatedElement(suffix.getValue().toString(),
+                  element, type, prefixesArray));
         }
+      } catch (MalformedURLException e) {
+        e.printStackTrace();
```

Comment:
you'll want to throw an annotation processing error here instead of just printing the stack trace